component/prometheus/scrape/scrape.go

```diff
@@ -236,13 +228,22 @@ func (c *Component) Update(args component.Arguments) error {
 	defer c.mut.Unlock()
 	c.args = newArgs
 
+	// Update scraper with scrape options
+	data, err := c.opts.GetServiceData(http.ServiceName)
+	if err != nil {
+		return fmt.Errorf("failed to get information about HTTP server: %w", err)
+	}
+	newFlowAppendables, newScraper := c.createPromScrapeResources(data.(http.Data), newArgs)
+	c.appendable = newFlowAppendables
```

Comment:
I don't think we have to recreate the appendables; the call to UpdateChildren on line 240 should be sufficient, and we can return to having only one instance of `c.appendable` for the lifetime of the component.

Reply:
Sounds great, -1 work for me then 😄